TST: The config value graphviz_dot_args has type list, defaults to tuple

Looks like a compatibility patch is needed w.r.t. sphinxdev

```
build finished with problems, 1 warning (with warnings treated as errors).
----------------------------- Captured stderr call -----------------------------
WARNING: The config value `graphviz_dot_args' has type `list', defaults to `tuple'.
=========================== short test summary info ============================
FAILED sphinx_astropy/tests/test_conf_v1.py::test_conf - assert 1 == 0
FAILED sphinx_astropy/tests/test_conf_v1.py::test_intersphinx_toggle - ...
```

--- a/sphinx_astropy/conf/v1.py
+++ b/sphinx_astropy/conf/v1.py
@@ -15,6 +15,8 @@ from collections import ChainMap
 from os import path
 
 import astropy_sphinx_theme
+import sphinx
+from packaging.version import Version
 
 try:
     import astropy
@@ -33,7 +35,7 @@ else:
 # specific version check, call check_sphinx_version("x.y.z.") from
 # your project's conf.py
 needs_sphinx = '3.0'
-
+SPHINX_LT_8_2 = Version(sphinx.__version__) < Version("8.2.dev")
 
 on_rtd = os.environ.get('READTHEDOCS', None) == 'True'
 
@@ -246,6 +248,9 @@ graphviz_dot_args = (
     '-Gfontsize=10',
     '-Gfontname=Helvetica Neue, Helvetica, Arial, sans-serif',
 )
+
+if SPHINX_LT_8_2:
+    graphviz_dot_args = list(graphviz_dot_args)
 
 # -- Options for HTML output -------------------------------------------------
 

--- a/sphinx_astropy/conf/v1.py
+++ b/sphinx_astropy/conf/v1.py
@@ -238,14 +238,14 @@ autoclass_content = "both"
 # Render inheritance diagrams in SVG
 graphviz_output_format = "svg"
 
-graphviz_dot_args = [
+graphviz_dot_args = (
     '-Nfontsize=10',
     '-Nfontname=Helvetica Neue, Helvetica, Arial, sans-serif',
     '-Efontsize=10',
     '-Efontname=Helvetica Neue, Helvetica, Arial, sans-serif',
     '-Gfontsize=10',
     '-Gfontname=Helvetica Neue, Helvetica, Arial, sans-serif',
-]
+)
 
 # -- Options for HTML output -------------------------------------------------
 

--- a/sphinx_astropy/conf/v2.py
+++ b/sphinx_astropy/conf/v2.py
@@ -13,6 +13,9 @@ import os
 from collections import ChainMap
 from os import path
 
+import sphinx
+from packaging.version import Version
+
 try:
     import astropy
 except ImportError:
@@ -28,6 +31,7 @@ else:
 # The version check in Sphinx itself can only compare the major and
 # minor parts of the version number, not the micro.
 needs_sphinx = '4.2'
+SPHINX_LT_8_2 = Version(sphinx.__version__) < Version("8.2.dev")
 
 on_rtd = os.environ.get('READTHEDOCS', None) == 'True'
 
@@ -236,6 +240,9 @@ graphviz_dot_args = (
     '-Gfontsize=10',
     '-Gfontname=Helvetica Neue, Helvetica, Arial, sans-serif',
 )
+
+if SPHINX_LT_8_2:
+    graphviz_dot_args = list(graphviz_dot_args)
 
 # For sphinx-copybutton
 copybutton_prompt_text = r">>> |\.\.\. |\$ |In \[\d*\]: | {2,5}\.\.\.: | {5,8}: "

--- a/sphinx_astropy/conf/v2.py
+++ b/sphinx_astropy/conf/v2.py
@@ -228,14 +228,14 @@ autoclass_content = "both"
 # Render inheritance diagrams in SVG
 graphviz_output_format = "svg"
 
-graphviz_dot_args = [
+graphviz_dot_args = (
     '-Nfontsize=10',
     '-Nfontname=Helvetica Neue, Helvetica, Arial, sans-serif',
     '-Efontsize=10',
     '-Efontname=Helvetica Neue, Helvetica, Arial, sans-serif',
     '-Gfontsize=10',
     '-Gfontname=Helvetica Neue, Helvetica, Arial, sans-serif',
-]
+)
 
 # For sphinx-copybutton
 copybutton_prompt_text = r">>> |\.\.\. |\$ |In \[\d*\]: | {2,5}\.\.\.: | {5,8}: "

